include/deal.II/matrix_free/shape_info.h

```diff
@@ -356,21 +356,22 @@ namespace internal
        */
       dealii::Table<2, unsigned int> face_to_cell_index_hermite;
 
-      /**
-       * Check whether we have symmetries in the shape values. In that case,
-       * also fill the shape_???_eo fields.
-       */
-      bool
-      check_1d_shapes_symmetric(const unsigned int n_q_points_1d);
-
       /**
        * Check whether symmetric 1D basis functions are such that the shape
        * values form a diagonal matrix, i.e., the nodal points are collocated
        * with the quadrature points. This allows for specialized algorithms
        * that save some operations in the evaluation.
        */
       bool
-      check_1d_shapes_collocation();
+      check_1d_shapes_collocation() const;
```

Comment:
Can you make this one private as well?

Reply:
Actually, I would like to use this function externally for a constant object (which is the reason why I stumbled into this function).

Reply:
But wouldn't we then also possibly want to allow to look into the symmetric case with the `check_1d_shapes_symmetric()` call? It is on the same level of query. That would need some restructuring because we would need to separate the query about symmetric from filling additional `eo` fields (so the current naming is misleading already because it does not just check but also fill some fields).

But my actual question regarding public/private is that we have the result of those queries cached in the variable [element_type](https://github.com/dealii/dealii/blob/master/include/deal.II/matrix_free/shape_info.h#L146) with e.g. `tensor_symmetric_collocation` for the case of collocation. Did you look for that part?

Reply:
I overlooked the `ElementType`; I will rather use that. Then, I can make `check_1d_shapes_collocation` also private!